boards/stm32-based: model Kconfig

MrKevinWeiss opened the pull request:
### Contribution description

Fixes modelling in many (but not all) stm32 boards:

- alientek-pandora
- blackpill
- bluepill-128kib
- bluepill-stm32f030c8
- bluepill
- iotlab-a8-m3
- iotlab-m3
- lobaro-lorabox
- olimexino-stm32
- p-l496g-cell02
- stm32f429i-disco
- stm32f769i-disco
- stm32f723e-disco
- stm32l0538-disco
- stm32l476g-disco
- stm32mp157c-dk2
- ublox-c030-u201
- weact-f401cc
- weact-f401ce
- weact-f411ce
- ublox-c030-u201

Some boards will require more attention but any relevant modeling is done:
- stm32mp157c-dk2 (clocks)
- stm32f3discovery (? size diff)
- stm32f030f4-demo (? size diff)
- pyboard (? size the same)
- nucleo-l011k4 (picolib)
- lora-e5-dev (clocks)
- nucleo-wl55jc(clocks)

Some drivers needed to be touched to allow saul_default to select them:

- isl29020
- l3g4200d
- lm75
- lpsxxx
- lsm303dlhc
- saul

### Testing procedure

 Green CI and a look over if it makes sense

### Issues/PRs references

Part of #16875
Depends on #17376

MrKevinWeiss commented on boards/stm32f429i-disco/Kconfig:
```diff
@@ -13,20 +13,8 @@ config BOARD_STM32F429I_DISCO
-    select BOARD_HAS_LSE
 
 source "$(RIOTBOARD)/common/stm32/Kconfig"
+source "$(RIOTBOARD)/stm32f429i-disc1/Kconfig"
```
Note that this is not the normal way we do it, usually this would be factored into a common board or something instead of just taking another board...  This is what is done in the makefile and TBH, we should probably deprecate this board as it is super obsolete.

leandrolanzieri replied:
See #17376

fjmolinas replied:
@aabadie do you have the old or new version?

leandrolanzieri commented on boards/common/weact-f4x1cx/Makefile.features:
```diff
@@ -15,3 +15,9 @@ FEATURES_PROVIDED += highlevel_stdio
+
+# This configuration enables modules that are only available when using Kconfig
+# module modelling
+ifeq (1, $(TEST_KCONFIG))
```
See #17376

MrKevinWeiss commented:
with #17376 the picolib issues can be resolved and I  can add the nucleo-l011k4 and stm32f030f4-demo to the passing list.

MrKevinWeiss commented:
Hmmm it seems that the current forcing of 
```
# all cortex MCU's use newlib as libc
CONFIG_MODULE_NEWLIB=y

# use the nano-specs of Newlib when available
CONFIG_MODULE_NEWLIB_NANO=y
```

in `cpu/cortexm_common/cortexm_common.config` prevents the picolib from being selected from the board because it must select these things.  Maybe I can try to override the choice instead...

MrKevinWeiss commented:
I think the choice thing is the way to go.

I now override the libc_implementation in the nucleo-l011k4 and stm32f030f4-demo and got rid of the hardcoding in the cortex common.

I am running the CI to see all failing stm boards, some I know will fail I just want an overview.  Maybe I can get a few more in this PR but it might also be worth fixing what we have now and doing a follow-up later.  Possibly after another check on all boards.

MrKevinWeiss commented:
OK, another round.  I think I got all obvious things (only binary diffs with the same modules are left).  In the murdock file I indicated them with 2 spaces in front, but I guess this will be testing all boards anyways.
lora-e5-dev, nucleo-wl55jc, pyboard, stm32f3discovery, stm32mp157c-dk2.

Many of these are due to clock configurations which will be a different PR.

If everything is passes as expected I will clean things up, run another limited build and record the blacklisted boards and hopefully get this acked (then remove the limiting build commit)

MrKevinWeiss commented:
OK rebased and now I am only checking the boards that should pass. After the review/ack I will squash and remove the murdock change (though I may add some boards to the list).

fjmolinas reviewed (approved):
ACK, looks good, please squash if murdock passes the limited build

fjmolinas commented:
All green please squash!

MrKevinWeiss commented:
OK, squashed and added a few boards to the regular testing.  There may still be some errors in the few boards that were added as I only check a subset of apps.

We will see.

MrKevinWeiss commented:
It seems like some `periph_uart_fc_hw` is missing from the `makefile.features` for the ublox and stm32f7***-disco board.  I will add them in and directly squash.

MrKevinWeiss commented:
Everything looks good if we want to merge?

fjmolinas commented:
Gooo!
